From 130d1ad1b62ccadc23731dac5efa2f6749a9dcb9 Mon Sep 17 00:00:00 2001
From: "Richard A. Griffiths" <Richard.A.Griffiths@intel.com>
Date: Wed, 27 Nov 2013 15:04:42 -0800
Subject: [PATCH 144/429] thermal driver: compiler warnings pointer casts

correcting multiple casts of different sizes, removes unused variable
and sets -Werror in this driver's Makefile.
change -Werror to be file specific.

Signed-off-by: Richard A. Griffiths <Richard.A.Griffiths@intel.com>
---
 drivers/thermal/Makefile             |  4 ++++
 drivers/thermal/intel_mrfl_thermal.c | 13 ++++++-------
 drivers/thermal/intel_soc_thermal.c  | 12 ++++++------
 drivers/thermal/thermal_core.c       |  2 +-
 4 files changed, 17 insertions(+), 14 deletions(-)

diff --git a/drivers/thermal/Makefile b/drivers/thermal/Makefile
index 2d44fedd8cfc..86cb18cf5a50 100644
--- a/drivers/thermal/Makefile
+++ b/drivers/thermal/Makefile
@@ -2,6 +2,10 @@
 # Makefile for sensor chip drivers.
 #
 
+CFLAGS_intel_mrfl_thermal.o := -Werror
+CFLAGS_intel_soc_thermal.o := -Werror
+CFLAGS_thermal_core.o := -Werror
+
 obj-$(CONFIG_THERMAL)		+= thermal_sys.o
 thermal_sys-y			+= thermal_core.o
 
diff --git a/drivers/thermal/intel_mrfl_thermal.c b/drivers/thermal/intel_mrfl_thermal.c
index 3757743dc7bf..4d65a7e294ac 100644
--- a/drivers/thermal/intel_mrfl_thermal.c
+++ b/drivers/thermal/intel_mrfl_thermal.c
@@ -369,7 +369,7 @@ exit_err:
 	return ret;
 }
 
-static ssize_t store_trip_hyst(struct thermal_zone_device *tzd,
+static int store_trip_hyst(struct thermal_zone_device *tzd,
 				int trip, long hyst)
 {
 	int ret;
@@ -397,7 +397,7 @@ ipc_fail:
 	return ret;
 }
 
-static ssize_t show_trip_hyst(struct thermal_zone_device *tzd,
+static int show_trip_hyst(struct thermal_zone_device *tzd,
 				int trip, long *hyst)
 {
 	int ret;
@@ -416,7 +416,7 @@ static ssize_t show_trip_hyst(struct thermal_zone_device *tzd,
 	return ret;
 }
 
-static ssize_t store_trip_temp(struct thermal_zone_device *tzd,
+static int store_trip_temp(struct thermal_zone_device *tzd,
 				int trip, long trip_temp)
 {
 	int ret, adc_val;
@@ -443,7 +443,7 @@ exit:
 	return ret;
 }
 
-static ssize_t show_trip_temp(struct thermal_zone_device *tzd,
+static int show_trip_temp(struct thermal_zone_device *tzd,
 				int trip, long *trip_temp)
 {
 	int ret, adc_val;
@@ -470,7 +470,7 @@ exit:
 	return ret;
 }
 
-static ssize_t show_trip_type(struct thermal_zone_device *tzd,
+static int show_trip_type(struct thermal_zone_device *tzd,
 			int trip, enum thermal_trip_type *trip_type)
 {
 	/* All are passive trip points */
@@ -479,7 +479,7 @@ static ssize_t show_trip_type(struct thermal_zone_device *tzd,
 	return 0;
 }
 
-static ssize_t show_temp(struct thermal_zone_device *tzd, long *temp)
+static int show_temp(struct thermal_zone_device *tzd, long *temp)
 {
 	int ret;
 	struct thermal_device_info *td_info = tzd->devdata;
@@ -669,7 +669,6 @@ static struct thermal_zone_device_ops tzd_ops = {
 
 static int mrfl_thermal_probe(struct platform_device *pdev)
 {
-	struct device *dev = &pdev->dev;
 	int ret, i;
 	struct intel_mid_thermal_platform_data *pdata;
 
diff --git a/drivers/thermal/intel_soc_thermal.c b/drivers/thermal/intel_soc_thermal.c
index a32c7975cb3a..5849b6dadfaf 100644
--- a/drivers/thermal/intel_soc_thermal.c
+++ b/drivers/thermal/intel_soc_thermal.c
@@ -266,7 +266,7 @@ static void enable_soc_dts(void)
 	}
 }
 
-static ssize_t show_trip_hyst(struct thermal_zone_device *tzd,
+static int show_trip_hyst(struct thermal_zone_device *tzd,
 				int trip, long *hyst)
 {
 	u32 eax, edx;
@@ -289,7 +289,7 @@ static ssize_t show_trip_hyst(struct thermal_zone_device *tzd,
 	return 0;
 }
 
-static ssize_t store_trip_hyst(struct thermal_zone_device *tzd,
+static int store_trip_hyst(struct thermal_zone_device *tzd,
 				int trip, long hyst)
 {
 	u32 eax, edx;
@@ -314,7 +314,7 @@ static ssize_t store_trip_hyst(struct thermal_zone_device *tzd,
 	return 0;
 }
 
-static ssize_t show_temp(struct thermal_zone_device *tzd, long *temp)
+static int show_temp(struct thermal_zone_device *tzd, long *temp)
 {
 	struct thermal_device_info *td_info = tzd->devdata;
 	u32 val = read_soc_reg(PUNIT_TEMP_REG);
@@ -334,7 +334,7 @@ static ssize_t show_temp(struct thermal_zone_device *tzd, long *temp)
 	return 0;
 }
 
-static ssize_t show_trip_type(struct thermal_zone_device *tzd,
+static int show_trip_type(struct thermal_zone_device *tzd,
 			int trip, enum thermal_trip_type *trip_type)
 {
 	/* All are passive trip points */
@@ -343,7 +343,7 @@ static ssize_t show_trip_type(struct thermal_zone_device *tzd,
 	return 0;
 }
 
-static ssize_t show_trip_temp(struct thermal_zone_device *tzd,
+static int show_trip_temp(struct thermal_zone_device *tzd,
 				int trip, long *trip_temp)
 {
 	u32 aux_value = read_soc_reg(PUNIT_AUX_REG);
@@ -360,7 +360,7 @@ static ssize_t show_trip_temp(struct thermal_zone_device *tzd,
 	return 0;
 }
 
-static ssize_t store_trip_temp(struct thermal_zone_device *tzd,
+static int store_trip_temp(struct thermal_zone_device *tzd,
 				int trip, long trip_temp)
 {
 	u32 aux_trip, aux = 0;
diff --git a/drivers/thermal/thermal_core.c b/drivers/thermal/thermal_core.c
index 39d042646a72..65cb6056fab2 100644
--- a/drivers/thermal/thermal_core.c
+++ b/drivers/thermal/thermal_core.c
@@ -815,7 +815,7 @@ thermal_cooling_device_force_state_override_store(struct device *dev,
 	int ret;
 	struct thermal_cooling_device *cdev = to_cooling_device(dev);
 
-	ret = cdev->ops->set_force_state_override(cdev, buf);
+	ret = cdev->ops->set_force_state_override(cdev, (char *) buf);
 	if (ret)
 		return ret;
 
-- 
2.37.3

